### ISSUE TITLE
Network issues on  release 2022.7.0 (and HA 2022.8 dev)

### ISSUE THREAD

**Issue by Mariusthvdb** [music-assistant/hass-music-assistant#717]
### What version of Music Assistant has the issue?

2022.7.0

### The problem

experiencing awful Nabu Casa cloud connection ever since 2022.7 is installed. Also seeing more pychromecast errors than before

Also, it wont tts my status message, stating it isnt found in cache. That's correct, I dont use cache on TTS and never did, have that disabled.

### How to reproduce

just open the cloud url and experience practically no view fully loading. When opening a camera images view, it wont load at all, nor release the browser...

### Relevant log output

network issues:
```
Logger: async_upnp_client.ssdp
Source: components/ssdp/__init__.py:341 
First occurred: 23:02:46 (16 occurrences) 
Last logged: 23:32:47

Received error: [Errno 101] Network unreachable, transport: <_SelectorDatagramTransport fd=16 read=polling write=<idle, bufsize=0>>
```

tts issues:
```
2022-07-23 23:37:33.653 ERROR (MainThread) [homeassistant.components.tts] Error on load tts: 55819e57f99be072c2997990d0b0dc7c5e838932_nl-nl_826d21f2a0_cloud not in cache!
2022-07-23 23:37:33.677 WARNING (MainThread) [music_assistant.players.stream] Stream error on http://192.168.1.61:42938/api/tts_proxy/55819e57f99be072c2997990d0b0dc7c5e838932_nl-nl_826d21f2a0_cloud.mp3
2022-07-23 23:37:33.682 ERROR (MainThread) [homeassistant.components.tts] Error on load tts: 07d3515add3514bf29c79533727dc799c1fde2e8_nl-nl_826d21f2a0_cloud not in cache!
2022-07-23 23:37:33.700 WARNING (MainThread) [music_assistant.players.stream] Stream error on http://192.168.1.61:42938/api/tts_proxy/07d3515add3514bf29c79533727dc799c1fde2e8_nl-nl_826d21f2a0_cloud.mp3
2022-07-23 23:37:42.911 WARNING (SyncWorker_1) [pychromecast.controllers] Quick Play failed for http://192.168.1.61:42938/api/tts_proxy/6a9ea9b12ab5902c628320c6f39c8c6b8be08739_nl-nl_826d21f2a0_cloud.mp3:audio/mpeg({})
2022-07-23 23:37:42.923 ERROR (MainThread) [homeassistant.core] Error executing service: <ServiceCall media_player.play_media (c:01G8PFWTE3BT4P2P7BTAR4AHMY): entity_id=['media_player.googlehome_hub'], announce=True, extra=, media_type=music, media_id=media-source://tts/cloud?message=GoedeAvond,+Home-assistant+Rpi4+is+weer+actief+sinds+Zaterdag+23+Juli+en+buiten+is+het+18.0+graden.&language=nl-NL&gender=male>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 1731, in catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1750, in _execute_service
    await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
  File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 204, in handle_service
    await service.entity_service_call(
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 676, in entity_service_call
    future.result()  # pop exception if have
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 930, in async_request_call
    await coro
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 713, in _handle_entity_call
    await result
  File "/usr/src/homeassistant/homeassistant/components/cast/media_player.py", line 737, in async_play_media
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/quick_play.py", line 77, in quick_play
    controller.quick_play(**data)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/controllers/media.py", line 547, in quick_play
    raise PyChromecastError()
pychromecast.error.PyChromecastError
2022-07-23 23:37:51.956 WARNING (SyncWorker_10) [pychromecast.controllers] Quick Play failed for http://192.168.1.61:8095/ebb4b47cb9324328906a061393cddffd.flac:audio/flac({'enqueue': False, 'stream_type': 'BUFFERED', 'title': ' Streaming from Music Assistant'})
2022-07-23 23:37:52.061 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 537, in play_index
    await self.player.play_url(stream.url)
  File "/config/custom_components/mass/player_controls.py", line 535, in play_url
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/quick_play.py", line 77, in quick_play
    controller.quick_play(**data)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/controllers/media.py", line 547, in quick_play
    raise PyChromecastError()
pychromecast.error.PyChromecastError
2022-07-23 23:37:54.201 WARNING (SyncWorker_15) [pychromecast.controllers] Quick Play failed for http://192.168.1.61:8095/e64605a573fe424db9e22e7fe9da6c61.flac:audio/flac({'enqueue': False, 'stream_type': 'BUFFERED', 'title': ' Streaming from Music Assistant'})
2022-07-23 23:37:54.255 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 537, in play_index
    await self.player.play_url(stream.url)
  File "/config/custom_components/mass/player_controls.py", line 535, in play_url
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/quick_play.py", line 77, in quick_play
    controller.quick_play(**data)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/controllers/media.py", line 547, in quick_play
    raise PyChromecastError()
pychromecast.error.PyChromecastError
2022-07-23 23:39:48.619 ERROR (Thread-8) [homeassistant.components.cast.media_player] Failed to cast media http://192.168.1.61:8095/ebb4b47cb9324328906a061393cddffd.flac. Please make sure the URL is: Reachable from the cast device and either a publicly resolvable hostname or an IP address
2022-07-23 23:39:48.822 ERROR (MainThread) [homeassistant.components.tts] Error on load tts: 55819e57f99be072c2997990d0b0dc7c5e838932_nl-nl_826d21f2a0_cloud not in cache!
2022-07-23 23:39:48.833 WARNING (MainThread) [music_assistant.players.stream] Stream error on http://192.168.1.61:42938/api/tts_proxy/55819e57f99be072c2997990d0b0dc7c5e838932_nl-nl_826d21f2a0_cloud.mp3
2022-07-23 23:40:06.098 WARNING (SyncWorker_15) [pychromecast.controllers] Quick Play failed for http://192.168.1.61:8095/c8b1072cc55c452a83ae9517e0f00759.flac:audio/flac({'enqueue': False, 'stream_type': 'BUFFERED', 'title': ' Streaming from Music Assistant'})
2022-07-23 23:40:06.108 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/music_assistant/models/player_queue.py", line 537, in play_index
    await self.player.play_url(stream.url)
  File "/config/custom_components/mass/player_controls.py", line 535, in play_url
    await self.hass.async_add_executor_job(
  File "/usr/local/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/quick_play.py", line 77, in quick_play
    controller.quick_play(**data)
  File "/usr/local/lib/python3.10/site-packages/pychromecast/controllers/media.py", line 547, in quick_play
    raise PyChromecastError()
pychromecast.error.PyChromecastError
```

### Additional information

aware I am using 2022.8 dev which has some issues with the players and also wont show the library. Had that before, and no problem waiting for the next beta release cycle.

The network issues however are new, and practically fatal to a Cloud connected instance.

So I disabled MA for now, and all came back.

### What version of Home Assistant Core are your running

2022.8.0.dev20220722

### What type of installation are you running?

Home Assistant OS

### On what type of hardware are you running?

Generic x86-64 (e.g. Intel NUC)

**Comment by marcelveldt** [comment]
The Nabu Casa cloud connection has nothing to do with MA so that is a real mystery but again you're running a DEV version of HA supposed for development purposes and not running on your own setup ;-)

As for the cast issues, I'm seeing those myself too and today I ran a number of tests and I think this is just an issue in the cast integration (or underlying pychromecast lib) that gets triggered by MA. See the other open issue for that.

I will release a first beta for 2022.8 on monday I think or maybe tomorrow with a first HA 2022.8 compatibility

**Comment by Mariusthvdb** [comment]
as said: aware of 2022.8 dev and previously mentioned dev vs production. Not to insist or anything, and calmly awaiting your updates ;-)

this however is new, and reproducible.

Ive rebooted all, even had Ziggo servicedesk reboot my cable modem.

just opened the nabucasa link once again, and it seems to load initially, albeit even slower than usual (NC is way slower than duckdns).
After pulling in my camera images, the views can not be changed anymore, it practically hangs the system.

speedtest is alright, so no internet/lan issues

<img width="1039" alt="Schermafbeelding 2022-07-24 om 14 24 28" src="https://user-images.githubusercontent.com/33354141/180646871-f779691a-c9ca-441c-b00e-711c97768540.png">

Taking out Music Assistant get things going once again...

**Comment by erkr** [comment]
With 2022.7.0 the pending errors are back as well.
I know not directly MA, but indicates high stress on the system:
``` 

[546747333888] Client exceeded max pending messages [2]: 2048
19:45:28 – (ERROR) Home Assistant WebSocket API - message first occurred at 19:13:43 and shows up 5693 times
[546747333888] Client unable to keep up with pending messages. Stayed over 512 for 5 seconds
19:45:28 – (ERROR) Home Assistant WebSocket API - message first occurred at 19:05:15 and shows up 480 times
```

**Comment by marcelveldt** [comment]
Is this perhaps during the initial scan ? because at that stage A LOT of messages are sent out so that could explain this behavior. A simple fix from my side would be not to notify the frontend of media changes when the full scan runs.

**Comment by Mariusthvdb** [comment]
Fwiw, in my config, nothing is scanned, I do see a .db file of about 20mb, so it’s not deleted either, but in the end, there’s nothing in the library to play.

If that fix wouldn’t cloud any real trouble, it might be well worth it.

ofc we need to know when things are amiss, so lowering the logging isn’t always a good idea…?

**Comment by marcelveldt** [comment]
This is not the logging but its the backend informing the frontend (over websockets) that a media item was changed.
Very theoretical though because it means that you have your Music Assistant panel opened while doing the initial scan.

I'm still blaming cast for all these issues as I've seen all kind of strange issues only when cast players were involved.
As a test, can you remove the cast players from the MA config (maybe create backup first) and then see if the issue still exists with Nabu Casa cloud ?

**Comment by marcelveldt** [comment]
> With 2022.7.0 the pending errors are back as well. I know not directly MA, but indicates high stress on the system:

Not exactly... Those errors can occur i f there are simply too many messages (e.g. when you have too many devices sending state changes for example or MA initial scan shouting a lot of "media changed!" events.
But... you need to have A LOT of messages to make that happen or a really slow client device.

This error can also occur if something is locking up the eventloop, even for a short while.
Home Assistant (and Music Assistant) is written in async python but sometimes do need to interact with oldskool synchronous libraries or functions, such as Google cast. If such interaction is not properly executed in a "executor", this sync code will temporary block the main eventloop and thus stalling everything that is happening.

Both HA core and MA core is 100% async code only and the small amount of sync code is run in an executor.
So, my feeling is that there is some component commonly in HA/MA with an undetected I/O lock issue. The common identifier so far seems to be cast. 

A very clear indicator of my feeling sits in that "pychromecast.error.PyChromecastError" where the chromecast was not able to play music from the stream sent to it. At that time the pychromecast itself locked the eventloop causing MA to fail to stream. A chicken and the egg situation.

**Comment by erkr** [comment]
I fully recognize your reasoning! Not 100% sure, but I believe there indeed was always an "pychromecast.error.PyChromecastError" before the lending errors pop up

**Comment by marcelveldt** [comment]
In that case I think I'll continue the hunt on the cast road. Besides that it would be great if one of you could run a test with no cast players enabled in HA.

**Comment by Mariusthvdb** [comment]
no cast players in HA, or in MA....
In my config, that would leave only the Symfonisk, which is ok I guess, and the Apple HomePod, which is an issue in itself..

**Comment by erkr** [comment]
I will try that with my test setup, but I don't have other players except the local vlc telnet

**Comment by marcelveldt** [comment]
@Mariusthvdb I am very, very curious if skipping the cast players resolves the strange connection issue you see with Nabu Casa.

**Comment by marcelveldt** [comment]
> I will try that with my test setup, but I don't have other players except the local vlc telnet

you may want to try squeezelite on your device ?
just download squeezelite, setup slimproto integration in HA and it will work out of the box

**Comment by Mariusthvdb** [comment]
right, so, if I delete them from the MA config, and keep the HomePod and Symfonisk, that would be a good test?

I have MA disabled currently, and all cast players are available in HA, and the NC issue is not there.
let me try.

**Comment by Mariusthvdb** [comment]
initial feedback after taking out cast devices:

- no more errors as posted above (no use either of those devices, so not sure if thats really helpful)
- using the NC external link now keeps an acceptable speed of the frontend, And allows me to show the cameras and navigate away from those
- after having also deleted the music assistant.db (hoped it would rewrite a correct one), I can indeed it being created and growing. However, still nothing in the frontend is displayed. (which is probably due to my 2022.8 dev install?)

hope this helps a bit

**Comment by marcelveldt** [comment]
@Mariusthvdb see #713 as meta issue for the google cast problems.
I think I found a/the trigger and placed a small workaround in place. It's not the root cause nor fix but at least the issue is no longer triggered by MA in the cast integration. I can work from here with Eric (cast maintainer) to solve the real underlying issue.

Can you give the 2022.7.1 patch version a try. I'm curious if it fixes your cast and TTS issues and if it also fixes that nabu casa connection issue I don't understand.

**Comment by marcelveldt** [comment]
Figured out the root cause for the connectivity issues and fixed it in 2022.7.2 (and 2022.8 beta 0).
